apps/src/lab2/views/Lab2ShareDialogWrapper.tsx

```diff
@@ -51,22 +60,57 @@ const Lab2ShareDialogWrapper: React.FunctionComponent<
   const isPublished = false;
   const canShareSocial = isSignedIn && is13Plus;
 
+  const dispatch = useAppDispatch();
+  const [submissionStatus, setSubmissionStatus] = useState<
+    SubmissionStatusType | undefined
+  >(undefined);
+
+  useEffect(() => {
+    const fetchStatus = async () => {
+      const response = await getSubmissionStatus();
+      setSubmissionStatus(response?.status);
+    };
+    fetchStatus();
```

Comment:
nit: might be simpler to just use promise syntax? i.e. `getSubmissionStatus().then(response => setSubmissionStatus(response.status))`

Reply:
Also, optional suggestion, but thinking about how we might use the Submit to be featured in other dialogs - we could potentially roll in the fetching of the submission status into the button component itself, and have it manage its own UI state in terms of showing, hiding, loading spinners, etc. That way when we want to put that button on another dialog we can just use the component directly. Not required, but just a thought to consolidate related code.

Reply:
Yes, I was planning to handle the 'Submit' button state in a follow-up while project is being submitted and what it should look like afterwards (an alert? a different label on a disabled button?,...) I'll check in with prod/design about this next step soon.

Reply:
Chatted offline! @sanchitmalhotra126 was referring to only the 'Submit to be featured' button in the `ShareDialog`, and I was thinking about the 'Submit' button in the `SubmitProjectDialog`.
When I moved the fetching of submission status to the `SubmitButtonInfo` internal component, there was a delay in the rendering of button every time the share modal was displayed. 


https://github.com/user-attachments/assets/2d91dcd3-51c7-4e94-ba5a-31a7522b0b88



So keeping the fetching in the parent component for now.